components/dm/ansible/import.go

```diff
@@ -0,0 +1,618 @@
+	"gopkg.in/yaml.v2"
+)
+
+// ref https://docs.ansible.com/ansible/latest/reference_appendices/config.html#the-configuration-file
+// Changes can be made and used in a configuration file which will be searched for in the following order:
+//
+// ANSIBLE_CONFIG (environment variable if set)
+// ansible.cfg (in the current directory)
+// ~/.ansible.cfg (in the home directory)
+// /etc/ansible/ansible.cfg
+func searchConfigFile(dir string) (fname string, err error) {
+	// ANSIBLE_CONFIG (environment variable if set)
+	if v := os.Getenv("ANSIBLE_CONFIG"); len(v) > 0 {
+		return v, nil
+	}
+
+	// ansible.cfg (in the current directory)
+	f := filepath.Join(dir, "ansible.cfg")
+	if utils.IsExist(f) {
+		return f, nil
+	}
+
+	// ~/.ansible.cfg (in the home directory)
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", errors.AddStack(err)
+	}
+	f = filepath.Join(home, ".ansible.cfg")
+	if utils.IsExist(f) {
+		return f, nil
+	}
+
+	// /etc/ansible/ansible.cfg
+	f = "/etc/ansible/ansible.cfg"
+	if utils.IsExist(f) {
+		return f, nil
+	}
+
+	return "", errors.Errorf("can not found ansible.cfg, dir: %s", dir)
+}
+
+func readConfigFile(dir string) (file *ini.File, err error) {
+	fname, err := searchConfigFile(dir)
+	if err != nil {
+		return nil, errors.AddStack(err)
+	}
+
+	file, err = ini.Load(fname)
+	if err != nil {
+		return nil, errors.Annotatef(err, "failed to load ini: %s", fname)
+	}
+
+	return
+}
+
+func firstNonEmpty(ss ...string) string {
+	for _, s := range ss {
+		if s != "" {
+			return s
+		}
+	}
+
+	return ""
+}
+
+func getAbsPath(dir string, path string) string {
+	if filepath.IsAbs(path) {
+		return path
+	}
+
+	path = filepath.Join(dir, path)
+	return path
+}
+
+// ExecutorGetter get the executor by host.
+type ExecutorGetter interface {
+	Get(host string) (e executor.Executor)
+}
+
+// Importer used for import from ansible.
+// ref DM docs: https://docs.pingcap.com/zh/tidb-data-migration/dev/deploy-a-dm-cluster-using-ansible
+type Importer struct {
+	dir               string // ansible directory.
+	inventoryFileName string
+	native            bool
+	sshTimeout        int64
+
+	// following vars parse from ansbile
+	user    string
+	sources map[string]*SourceConfig // addr(ip:port) -> SourceConfig
+
+	// only use for test.
+	// when setted, we use this executor instead of getting a truly one.
+	testExecutorGetter ExecutorGetter
+}
+
+// NewImporter create an Importer.
+func NewImporter(ansibleDir string, inventoryFileName string) (*Importer, error) {
+	dir, err := filepath.Abs(ansibleDir)
+	if err != nil {
+		return nil, errors.AddStack(err)
+	}
+
+	return &Importer{
+		dir:               dir,
+		inventoryFileName: inventoryFileName,
+		sources:           make(map[string]*SourceConfig),
+	}, nil
+}
+
+func (im *Importer) getExecutor(host string, port int) (e executor.Executor) {
+	if im.testExecutorGetter != nil {
+		return im.testExecutorGetter.Get(host)
+	}
+
+	keypath := ansible.SSHKeyPath()
+
+	cfg := executor.SSHConfig{
```

Comment:
Do we need to support native ssh?

Reply:
added https://github.com/pingcap/tiup/pull/672/commits/bfc8fb609c90ee20a2e9a00bb0366d1b14560fe2